pkgs/development/compilers/llvm/10/openmp/default.nix

```diff
@@ -15,7 +16,9 @@ stdenv.mkDerivation rec {
   src = fetch pname "0i4bn84lkpm5w3qkpvwm5z6jdj8fynp7d3bcasa1xyq4is6757yi";
 
   nativeBuildInputs = [ cmake perl ];
-  buildInputs = [ llvm ];
+  buildInputs = [
+    (if stdenv.buildPlatform == stdenv.hostPlatform then llvm else targetLlvm)
```

Comment:
(Commenting only on this one particular instance, but obviously applies to all.)

When buildPlatform == hostPlatform is there a meaningful difference between llvm and targetLlvm? In other words, could we always use targetLlvm here?

If this is done only to avoid rebuilds, I'd personally prefer rebuilds to long term code complexity / "technical debt".

Reply:
Here's why: if I simply use targetPackges, then `pkgsCross.aarch64-multiplatform.buildPackages.llvmPackages_14.openmp` will result in openmp built for the wrong architecture. I admit I'm not sure why this is the case, and there may be a better way to model this.